contracts/Booster.sol

```diff
@@ -190,34 +202,39 @@ contract Booster is ReentrancyGuardUpgradeable {
 
         //reward contracts are immutable or else the owner
         //has a means to redeploy and mint cvx via rewardClaimed()
-        if (lockRewards == address(0)) {
-            require(_rewards != address(0), "Not allowed!");
+        if (lockRewards == address(0) && _rewards != address(0)) {
             lockRewards = _rewards;
         }
-        if (stakerRewards == address(0)) {
-            require(_stakerRewards != address(0), "Not allowed!");
+        if (stakerRewards == address(0) && _stakerRewards != address(0)) {
             stakerRewards = _stakerRewards;
         }
-        if (stakerLockRewards == address(0)) {
-            require(_stakerLockRewards != address(0), "Not allowed!");
+        if (stakerLockRewards == address(0) && _stakerLockRewards != address(0)) {
             stakerLockRewards = _stakerLockRewards;
         }
 
         emit RewardContractsUpdated(_rewards, _stakerRewards, _stakerLockRewards);
     }
 
-    // Set reward token and claim contract, get from Curve's registry
-    function setFeeInfo(uint256 _lockFeesIncentive, uint256 _stakerLockFeesIncentive) external {
+    // Set reward token and distro claim contract; create new virtual reward pool and sets execution hash
+    // Also used to add or update distro contracts and execution hashes for an already active reward token
+    function setFeeInfo(
+        uint256 _lockFeesIncentive,
+        uint256 _stakerLockFeesIncentive,
+        address _feeToken,
+        address _distro,
+        bytes32 _executionHash
+        ) external {
         require(msg.sender == feeManager, "!auth");
-        require(_lockFeesIncentive.add(_stakerLockFeesIncentive) == FEE_DENOMINATOR);
+        require(_lockFeesIncentive.add(_stakerLockFeesIncentive) == FEE_DENOMINATOR, "!fee");
 
         lockFeesIncentive = _lockFeesIncentive;
         stakerLockFeesIncentive = _stakerLockFeesIncentive;
 
-        address _feeToken = IFeeDistro(feeDistro).token();
-        if (feeToken != _feeToken) {
+        if (feeTokens[_feeToken].active != true) {
+            require(!gaugeMap[_feeToken], "!token");
+            //require that we initialize at the zero index
             //create a new reward contract for the new token
-            lockFees = IRewardFactory(rewardFactory).CreateTokenRewards(_feeToken, lockRewards);
+            address lockFees = IRewardFactory(rewardFactory).CreateTokenRewards(_feeToken, lockRewards);
 
             if (_feeToken != veAsset) {
                 IRewards(stakerLockRewards).addReward(
```

Comment:
it will create new reward pool (lookFees) for each fee tokens and use mapping to save it